addons/l10n_es_edi_facturae_adm_centers/tests/test_edi_xml.py

```diff
@@ -0,0 +1,59 @@
+from hashlib import sha1
+from unittest.mock import patch
+
+import lxml
+from freezegun.api import freeze_time
+
+from odoo.fields import Command
+from odoo.tests import tagged
+from odoo.tools import file_open
+
+from odoo.addons.l10n_es_edi_facturae.tests.test_edi_xml import TestEdiFacturaeXmls
+
+
+@tagged('post_install_l10n', 'post_install', '-at_install')
+class TestEdiFacturaeAdmCentersXmls(TestEdiFacturaeXmls):
+    @classmethod
+    def setUpClass(cls, chart_template_ref='es_full', edi_format_ref=None):
+        super().setUpClass(chart_template_ref, edi_format_ref)
+
+        cls.partner_b.write({
+            'name': 'Ayuntamiento de San Sebastián de los Reyes',
+            'is_company': True,
+            'country_id': cls.env.ref('base.es').id,
+            'vat': 'P2813400E',
+            'city': 'San Sebastián de los Reyes',
+            'street': 'Plaza de la Constitución, 1',
+            'zip': '28701',
+            'state_id': cls.env.ref('base.state_es_m').id,
+        })
+        partner_b_ac = cls.partner_b.copy()
+        partner_b_ac.write({
+            'type': 'facturae_ac',
+            'parent_id': cls.partner_b.id,
+            'name': 'Intervención Municipal',
+            'l10n_es_edi_facturae_ac_center_code': 'L01281343',
+            'l10n_es_edi_facturae_ac_role_type_ids': [
+                Command.link(cls.env.ref('l10n_es_edi_facturae_adm_centers.ac_role_type_01').id),
+                Command.link(cls.env.ref('l10n_es_edi_facturae_adm_centers.ac_role_type_02').id),
+                Command.link(cls.env.ref('l10n_es_edi_facturae_adm_centers.ac_role_type_03').id),
+            ],
+        })
+
+    def test_generate_with_administrative_centers(self):
+        with freeze_time(self.frozen_today), \
+                patch(f"{self.certificate_module}.fields.datetime.now", lambda x=None: self.frozen_today), \
+                patch(f"{self.certificate_module}.sha1", lambda x: sha1()):
```

Comment:
This part was copied from the existing code in `l10n_es_edi_facturae`. I will check with @JulienAlardot, who wrote it in the original module.

Reply:
> Why don't we use freeze_time as a decorator of the method?
- We could

> Why use frozen_today? It is just making understanding more complex.
- I disagree, as the date itself doesn't matter as long as it's consistent

> Why patching datetime.now? freeze_time should do that?
- Because somehow it wasn't enough

> Why the self.certificate_module thing?
- Legacy of the account_edi dependency I forgot to rewrite

> How does the sha1 patch work? Why is it needed? Isn't the hashed string already fixed anyways? Can't we just use a __ignore__ in the test xml otherwise?
- That one's a bit blurry in my mind but it's a mix between the attributes not being able to be `__ignore__` when not in an attribute node and the export file system not generating them that way